master/internal/db/postgres_trial_logs.go

```diff
@@ -42,5 +47,69 @@ ORDER BY l.id ASC OFFSET $2 LIMIT $3
 `, fragment)
 
 	var b []*model.TrialLog
-	return b, db.queryRows(query, &b, params...)
+	return b, nil, db.queryRows(query, &b, params...)
+}
+
+// AddTrialLogs adds a list of *model.TrialLog objects to the database with automatic IDs.
+func (db *PgDB) AddTrialLogs(logs []*model.TrialLog) error {
+	if len(logs) == 0 {
+		return nil
+	}
+
+	var text strings.Builder
+	text.WriteString(`
+INSERT INTO trial_logs
+  (trial_id, message, log, agent_id, container_id, rank_id, timestamp, level, stdtype, source)
+ VALUES
+`)
+
+	args := make([]interface{}, 0, len(logs)*10)
+
+	for i, log := range logs {
+		if i > 0 {
+			text.WriteString(",")
+		}
+		fmt.Fprintf(&text, " ($%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d)",
+			i*10+1, i*10+2, i*10+3, i*10+4, i*10+5, i*10+6, i*10+7, i*10+8, i*10+9, i*10+10)
+
+		var l *model.RawString
+		if log.Log != nil {
+			r := model.RawString(*log.Log)
+			l = &r
+		}
+
+		args = append(args, log.TrialID, log.Message, l, log.AgentID, log.ContainerID, log.RankID,
+			log.Timestamp, log.Level, log.StdType, log.Source)
+	}
+
+	if _, err := db.sql.Exec(text.String(), args...); err != nil {
+		return errors.Wrapf(err, "error inserting %d trial logs", len(logs))
+	}
+
+	return nil
+}
+
+// TrialLogCount returns the number of logs in postgres for the given trial.
+func (db *PgDB) TrialLogCount(trialID int, fs []api.Filter) (int, error) {
+	params := []interface{}{trialID}
+	fragment, params := filtersToSQL(fs, params)
+	query := fmt.Sprintf(`
+SELECT count(*)
+FROM trial_logs l
+WHERE l.trial_id = $1
+%s
+`, fragment)
+	var count int
+	err := db.sql.QueryRow(query, params...).Scan(&count)
+	if err != nil {
+		return 0, err
+	}
+	return count, err
+}
+
+// TrialLogFields returns the unique fields that can be filtered on for the given trial.
+func (db *PgDB) TrialLogFields(trialID int) (*apiv1.TrialLogsFieldsResponse, error) {
+	var fields *apiv1.TrialLogsFieldsResponse
+	err := db.QueryProto("get_trial_log_fields", fields, trialID)
```

Comment:
This is (essentially) this same code in master and that works. I just moved it https://github.com/determined-ai/determined/blob/master/master/internal/api_trials.go#L189-L197.

Reply:
Under the hood, queryproto is just marshalling the resp into a `map[string]interface{}` then marshalling that into this, and i think this would work fine with JSON normally.